### ISSUE TITLE
Error when adding average value of metric column - Omniboard 2.4.3

### ISSUE THREAD

**Issue by serhiy-yevtushenko** [vivekratnavel/omniboard#170]
I got the following exception in omniboard log and undefined error, when trying to add metric column, which should show the average value:

omniboard_1  | MongoError: Unrecognized expression '$average'
omniboard_1  |     at Connection.<anonymous> (/usr/omniboard/node_modules/mongodb/lib/core/connection/pool.js:454:61)
omniboard_1  |     at Connection.emit (events.js:198:13)
omniboard_1  |     at processMessage (/usr/omniboard/node_modules/mongodb/lib/core/connection/connection.js:368:10)
omniboard_1  |     at Socket.<anonymous> (/usr/omniboard/node_modules/mongodb/lib/core/connection/connection.js:537:15)
omniboard_1  |     at Socket.emit (events.js:198:13)
omniboard_1  |     at addChunk (_stream_readable.js:287:12)
omniboard_1  |     at readableAddChunk (_stream_readable.js:268:11)
omniboard_1  |     at Socket.Readable.push (_stream_readable.js:223:10)
omniboard_1  |     at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
omniboard_1  | Query took 0.003 seconds.

**Comment by vivekratnavel** [comment]
If you are using v2.x of Omniboard, then MongoDB >=v4.x is required. Which version of MongoDB are you connecting to?

**Comment by serhiy-yevtushenko** [comment]
Thank you for the reply

The error appeared when using omniboard from docker-compose. The reason for it that it their simply mongo image is specified without version, and on this machine the older version of omniboard with older mongodb was also used.

I would advise to specify minimum docker version in example docker compose configuration for omniboard

**Comment by lotsoffood** [comment]
I am getting the same error with Mongo db version v4.0.5, and Omniboard 2+. Do all experiments need to have the same metrics for this to work? In simpler terms, if only a few rows have "metricA" and I create a metric columns based on "metricA" will it still work? I'm thinking this may be causing it.

**Comment by serhiy-yevtushenko** [comment]
For me the error was fixed, when I added version 4.0.13 of mongo db explicitly to mongodb image specification in docker file. Having latest version was not enough, as docker cached other version under latest (previous one)

**Comment by lotsoffood** [comment]
I am running Omniboard 2.4.3 and Mongo DB 4.0.5 locally (i.e. not inside of docker containers). Is it possible that because I have multiple experiments which have different metrics (for example, some rows may have MetricA, and others only MetricB) that it would cause a problem when trying to add MetricA?
